File: app/src/main/java/com/example/pillnotifier/model/DataHolder.java
package com.example.pillnotifier.model;

import java.util.HashMap;

public class DataHolder {
    private static final HashMap<String, String> data = new HashMap<>();
    public static String getData(String key) {
        return data.get(key);
    }
    public static void setData(String key, String value) {
        data.put(key, value);
    }
    public static void removeData(String key) {
        data.remove(key);
    }
}

File: app/src/main/java/com/example/pillnotifier/model/DataHolder.kt
package com.example.pillnotifier.model;

import android.content.Context
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.util.HashMap;

object DataHolder {
    private var data = HashMap<String, String>()
    fun getData(key: String): String? {
        return data[key]
    }
    fun setData(key: String, value: String) {
        data[key] = value
    }
    fun removeData(key: String) {
        data.remove(key)
    }
    private const val dataHolderCacheFile = "dataHolderCacheFile"
    fun cache(context: Context) {
        ObjectOutputStream(
            context.openFileOutput(
                dataHolderCacheFile,
                Context.MODE_PRIVATE
            )
        ).writeObject(data)
    }
    fun uploadFromCache(context: Context) {
        if (context.fileList().contains(dataHolderCacheFile)) {
            data =
                ObjectInputStream(context.openFileInput(dataHolderCacheFile)).readObject() as HashMap<String, String>
        }
    }
}
